kong/plugins/prometheus/exporter.lua

```diff
@@ -146,6 +150,21 @@ local function collect()
                  "/metrics endpoint: ", err)
   end
 
+  -- upstream targets accessible?
+  local balancer = require "kong.runloop.balancer"
+  local upstreams_dict = balancer.get_all_upstreams()
+  for name, id in pairs(upstreams_dict) do
+      local health_info = balancer.get_upstream_health(id)
+      for target, status in pairts(health_info) do
+          if status == "HEALTHY" then
+              metrics.upstream_target_health:set(1, { target })
+          else
+              metrics.upstream_target_health:set(0, { target })
+              kong.log.err("Upstream target is Unhealthy", target)
```

Comment:
We should not log the target status here, it is already logged by the balancer when the health checker detects a status change.